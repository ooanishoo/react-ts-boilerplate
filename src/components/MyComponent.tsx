import React from 'react';

const MyComponent: React.FC = () => {
    return <div data-testid="container">This is a custom component</div>;
};

export default MyComponent;
