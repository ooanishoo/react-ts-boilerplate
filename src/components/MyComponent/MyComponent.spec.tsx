import { render } from '@testing-library/react';
import React from 'react';
import MyComponent from './MyComponent';

describe('<MyComponent/>', () => {
    test('should render without crashing', () => {
        const expectedOutput = 'This is a custom component';
        const { container, getByTestId } = render(<MyComponent />);
        expect(container).toMatchSnapshot();
        expect(getByTestId('container')).toHaveTextContent(expectedOutput);
    });
});
