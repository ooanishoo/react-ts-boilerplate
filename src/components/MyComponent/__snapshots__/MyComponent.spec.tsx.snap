// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyComponent/> should render without crashing 1`] = `
<div>
  <div
    data-testid="container"
  >
    This is a custom component
  </div>
</div>
`;
